File: oop/project0/Circle.java
package com.java.project0;

public class Circle {
    private double radius;
    private Point center;
    private static final double PI = 3.14;

    public Circle() {
        this.radius = 1;
        this.center = new Point();
    }

    public Circle(double radiusValue, Point centerPoint) {
        this.radius = radiusValue;
        this.center = centerPoint;
    }

    public void display() {
        System.out.print("center: ");
        center.display();
        System.out.println("and radius: " + radius);
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }

    public Point getCenter() {
        return center;
    }

    public double area() {
        return PI * radius * radius;
    }

    public double perimeter() {
        return 2 * PI * radius;
    }

    public boolean compare(Circle other) {
        return (this.radius == other.radius && this.center.compare(other.center));
    }

    public void enlarge(double value) {
        this.radius += value;
    }
}

File: oop/project1/Circle.cpp
#include "Circle.h"

#include <iostream>

Circle::Circle() : radius(1), center(Point()) {}

Circle::Circle(double radiusValue, const Point& centerPoint)
    : radius(radiusValue), center(centerPoint) {}

void Circle::display() const {
    std::cout << "center: ";
    center.display();
    std::cout << "and radius: " << radius << std::endl;
}

double Circle::getRadius() const {
    return radius;
}

void Circle::setRadius(double radius) {
    this->radius = radius;
}

Point Circle::getCenter() const {
    return center;
}

double Circle::area() const {
    return PI * radius * radius;
}

double Circle::perimeter() const {
    return 2 * PI * radius;
}

bool Circle::compare(const Circle& other) const {
    return radius == other.radius && center.compare(other.center);
}

void Circle::enlarge(double value) {
    radius += value;
}
